compiler/stable_mir/src/ty.rs

```diff
@@ -565,3 +565,24 @@ pub enum ImplPolarity {
     Negative,
     Reservation,
 }
+
+#[derive(Clone, Debug)]
+pub struct Instance {
+    pub def: InstanceDef,
+    pub args: GenericArgs,
```

Comment:
Maybe merge this into the `InstanceDef` variants? For shims this duplicates information with the `Ty` and having it get out of sync could potentially crash the compiler and if it doesn't crash would likely misbehave.